src/linear_congruential_random_generator.c

```diff
@@ -2,22 +2,24 @@
 
 #include "linear_congruential_random_generator.h"
 
+#include <stdint.h> 
+
 /* The routines below implements the linear congruential random
    number generator. In particular, the function
    "set_linear_congruential_generator_seed" sets the seed of the random
    generator, while the function "linear_congruential_random_generator"
    returns a pseudo-random real number in the range [0, 1.0]. */
 
 /* Variables and pointers declarations */
-int ISEED = 16.; /* Default value in case the user does not specify the seed */
+uint16_t ISEED = 16; /* Default value in case the user does not specify the seed */
+#define LINEAR_RAND_MAX pow(2, sizeof(ISEED) * 8)
+
 
-/* Set the seed for the random number generator */
 void set_linear_congruential_generator_seed(int num) { ISEED = num; }
```

Comment:
should this argument be `uint16_t  num` ?

Reply:
Sure, why not